sipplauncher/Run.py

```
@@ -163,8 +163,15 @@ def run(args):
                 # 3. TestA run + TestB run
                 # 4. TestB post-run. OptionA: ValueB -> ValueA
                 # 5. TestA post-run: OptionB: ValueA -> ValueOrig
+                raise_exception = None
```

This shouldn't be a list? In case of multiple tasks ending in an exception...

---

Nacho, in my understanding we can pass to the upper layer only 1 exception.
Currently we're passing the last exception.
So all other exceptions from the list will be unused.
Therefore, I don't see why should we store them, if they will be unused anyway.

---

The problem I see is that we are returning just the last one, so we may missing important information. Can we at least log all exceptions? (if we are not logging them in other place).

---

Nacho, according to Danny's comment I changed the implementation to return just the first exception.

Regarding your comment, most of exceptions is already logged in test's run directory (try-except block inside SIPpTest.post_run()).
However some exceptions could be swallowed and hidden indeed (which don't hit the try-except block inside SIPpTest.post_run()).

The most simple solution would be to log the exception caught in Run.run() to main log as well.
This way, we'll store all exceptions in log, but for some cases there could be duplicates (same exception saved in test's run directory log + in main log).
To eliminate this logging dupication will need to have quite a big code change.
IMO it's not worth it.

So I propose just to add exception logging inside Run.run().